### ISSUE TITLE
Cannot import modules from `node_modules` within library definitions

### ISSUE THREAD

**Issue by Soreine** [facebook/flow#7137]
# TL;DR

In a library definition of mine, I'm able to import types from other library definitions but not modules from `node_modules`. [Reproducible example](https://github.com/Soreine/flow-libdefs-node-modules/tree/f46223ea0988a63bb2e43095126753b7fa09eb13)

# What I'm trying to do
While I have found a lot of issues regarding _importing types across library definitions_, in particular on the `flow-typed` repository, I have not found any discussion regarding **importing modules from `node_modules` inside a library definition**.

Importing types from modules or other libdefs is not even mentioned in the public documentation https://flow.org/en/docs/libdefs/creation/

I am using `immutable` and I am using a module (`some-module`) that also uses `immutable` data structures. I would like to write a library definition for `some-module`. But I cannot find a way to import `immutable` types from my library definition. `immutable` is inside `node_modules` and has types  packaged-in in the form of an `immutable.js.flow` file.

# Minimal reproducible example

Here is a minimal repo reproducing the problem https://github.com/Soreine/flow-libdefs-node-modules/tree/f46223ea0988a63bb2e43095126753b7fa09eb13

### `flow-typed/some-module.js`

```js
/* @flow */

declare module 'some-module' {
    // Error here `Cannot resolve module immutable`
    import type { List } from 'immutable';

    declare export var someList: List<number>;
}
```

### `test.js`

```js
/* @flow */
import { List } from 'immutable';
import { someList } from 'some-module';

// There should be an error here, but someList is typed as `any`
const wrong: string = someList;
```

I have even tried importing `immutable` using relative paths (`import type { List } from '../node_modules/immutable/dist/immutable'`) , but it's still not found.

The error is always as follow:

```
$ flow

Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ flow-typed/some-module.js:5:31

Cannot resolve module immutable.

     2│
     3│ declare module 'some-module' {
     4│     // Error here `Cannot resolve module immutable`
     5│     import type { List } from 'immutable';
     6│
     7│     declare var someList: List<number>;
     8│ }



Found 1 error
```

**Comment by Soreine** [comment]
Here is a relevant discussion about the different but closely related problem of importing types inside library definitions https://github.com/flow-typed/flow-typed/issues/2023

**Comment by SamChou19815** [comment]
This is intended behavior
